src/ImageSharp.Drawing.Text/GlyphBuilder.cs

```diff
@@ -0,0 +1,126 @@
+﻿// <copyright file="GlyphBuilder.cs" company="James Jackson-South">
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageSharp.Drawing
+{
+    using System.Collections.Generic;
+    using System.Numerics;
+
+    using SixLabors.Fonts;
+    using SixLabors.Shapes;
+
+    /// <summary>
+    /// rendering surface that Fonts can use to generate Shapes.
+    /// </summary>
+    internal class GlyphBuilder : IGlyphRenderer
+    {
+        private readonly PathBuilder builder = new PathBuilder();
+        private readonly List<IPath> paths = new List<IPath>();
+        private Vector2 currentPoint = default(Vector2);
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GlyphBuilder"/> class.
+        /// </summary>
+        public GlyphBuilder()
+            : this(Vector2.Zero)
+        {
+            // glyphs are renderd realative to bottom left so invert the Y axis to allow it to render on top left origin surface
+            this.builder = new PathBuilder();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GlyphBuilder"/> class.
+        /// </summary>
+        /// <param name="origin">The origin.</param>
+        public GlyphBuilder(Vector2 origin)
+        {
+            this.builder = new PathBuilder();
+            this.builder.SetOrigin(origin);
+        }
+
+        /// <summary>
+        /// Gets the paths that have been rendered by this.
+        /// </summary>
+        public IEnumerable<IPath> Paths => this.paths;
+
+        /// <summary>
+        /// Begins the glyph.
+        /// </summary>
+        void IGlyphRenderer.BeginGlyph()
```

Comment:
@tocsoft No, keep it as-is with the required cast. Best not to expose anything we don't explicitly want to.